lcls_naming_tool/lcls_naming_tool.py

```
@@ -0,0 +1,309 @@
+
+import argparse
+import subprocess
+import sys
+import json
+import re
+
+
+fc_dict = {}
+fg_dict = {}
+ccc_dict = {}
+beam_sources = {'K': '', 'L': ''}
+beam_numbers = {'0': '', '1': '', '2': '', '3': '', '4': '', '5': ''}
+
+
+def get_version():
+    result = subprocess.run(
+        ['git', 'describe', '--tags'], 
+        capture_output=True, 
+        text=True
+    )
+    return (str(result.stdout).strip().split('-'))[0]
+
+def display_version():
+    version = get_version()
+
+    parser = argparse.ArgumentParser(description="Current version of LCLS Naming Tool")
+    parser.add_argument('-v', '--version', action='store_true')
+    args = parser.parse_args()
+    
+    if args.version:
+        print(version)
+
+
+def load_taxons():
+    global fc_dict
+    global fg_dict
+    global ccc_dict
+
+    # Read the json files containing all the taxons
+    with open('taxons/functional_component_taxon.json') as fc_file:
+        fc_dict = json.load(fc_file)
+
+    with open('taxons/fungible_element_taxon.json') as fg_file:
+        fg_dict = json.load(fg_file)
+
+    with open('taxons/ccc_taxon.json') as ccc_file:
+        ccc_dict = json.load(ccc_file)
+
+
+def functional_component_is_valid(fc_taxon):
+
+    fc_length = len(fc_taxon)
+    
+    if (4 < len(fc_taxon) < 9):
+        fc_prefix = fc_taxon[:2]  # max length is currently 2 letters
+        fc_seq_num = fc_taxon[2:fc_length-2]  # max length set to 4
+        fc_source_ltr = fc_taxon[fc_length-2:fc_length-1] # max length is currently 1 letter
+        fc_beam_num = fc_taxon[fc_length-1:fc_length]     # max length is currently 1 digit
+    else:
+        return False
+
+    # validate FFFFF
+    try:
+        fc_dict[fc_prefix]
+        beam_sources[fc_source_ltr]
+        beam_numbers[fc_beam_num]
+
+        try:
+            int(fc_seq_num)
+
+            try:
+                1 / int(fc_seq_num[0])  # zero padding not allowed
+            except 0:
+                return False
+
+        except:
```

Bare except blocks catch system exceptions like `sys.exit` and `KeyboardInterrup` so they should be avoided. You should use the specific exceptions you're looking for if you know them, otherwise you can simply
```suggestion
        except Exception:
```